common/util.go

```diff
@@ -92,7 +92,7 @@ func HandleHTTPRequest(message string, headers http.Header, data ConnectorMetada
 	}
 
 	if resp.StatusCode < 200 && resp.StatusCode > 300 {
-		return nil, fmt.Errorf("request returned failure: %v. http_endpoint: %v, source: %v", resp.StatusCode, data.HTTPEndpoint, data.SourceName)
+		return resp, fmt.Errorf("request returned failure: %v. http_endpoint: %v, source: %v", resp.StatusCode, data.HTTPEndpoint, data.SourceName)
```

Comment:
@ranoble If status code isn't in that range, it's not a successful request and that's why I had returned `nil`. Can you please explain why this might be valid?

Reply:
To write the key, I need the key header. Event though the function fails, it still returns a valid result (just not the one we wanted). 
This could / would support distributed tracing, or the option to pass back useful debug info, so that the consumer of the error topic can take the correct action.

Reply:
Got it. In that case, can you please do the following?

1. Raise the PR in a separate branch as the common package is used by all connectors
2. Change the function return arguments to something better than it currently is. I am thinking status code, response and error. What are your thoughts?

Reply:
More than happy to. 
1. Sure - will do so. 
2. I'm a little more nervous about this one. This change is backwards compatible at the moment, `resp` can be safely ignored by any client. If someone is basing a custom connector off of this and upgrades that will cause a breaking change in their code. 

Happy to change it on your suggestion - but wanted to highlight that.

Reply:
That's true. Any change in the common package will have to be adopted by the remaining connectors . @therahulbhati 
What are your thoughts?

Reply:
Yes, @RealHarshThakur it's true. We should have this as a separate PR, we would also need to incorporate those changes for each connector as part of that PR.